Pool creation unknown tokens change price

1saf opened the pull request:
# Description

After some discussion with Pon, we came up with a solution that should also allow the user to change any token prices that they may have entered. This PR introduces a small 'Token Prices' card under Pool Summary which allows the user to go back and change any token prices they might've mistakenly, or had forgotten to enter.

The modal appearance behaviour has also changed. Instead of appearing immediately after selecting an unknown token, it will now appear on the next step as per design comments.


## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

## How should this be tested?
Here is a custom token: 0xbF3122A7a98a2F9971C339a8511378d853585876
Using an unknown token, add the  unknown token. It should appear on the side card with a warning to enter a price. 
You should be able to click the red 'Enter a price' row and it should bring up the unknown token price modal where you will be able to enter in a price. Test that this works properly

Another one to check is to make sure that the modal shows up on the next step (step 2: select fees) and applies the prices correctly there too.

Also please test mobile accordion behaviour.

## Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code where relevant, particularly in hard-to-understand areas
- [X] My changes generate no new console warnings
- [X] The base of this PR is `master` if hotfix, `develop` if not

### Walkthrough
(Note: I have no idea why USDC is returning 0)
https://www.loom.com/share/c1bcf8e807d04b2395fa2548e0ae1713

1saf commented on src/components/_global/BalInlineInput/BalInlineInput.vue:
```diff
@@ -119,17 +125,17 @@ function toggleEditable() {
             :type="type"
             :name="name"
-            :value="modelValue"
+            :value="format ? format(modelValue) : modelValue"
```
the format prop was not being used, this fixes that

1saf commented on src/components/_global/BalInlineInput/BalInlineInput.vue:
```diff
@@ -98,12 +98,18 @@ const inputAttrs = computed(() => omit(attrs, 'class'));
   }, 200);
 }
+
+function handleBlur(e: HtmlInputEvent) {
```
on blur of the component, toggling it to not editable seems to be the good UX move

garethfuller reviewed (approved):
Looks good. But why is the package-lock changing here?

1saf commented:
> Looks good. But why is the package-lock changing here?

Hmm, dunno. I'll kill the changes and try install again and that should fix it.

timjrobinson reviewed (approved):
Works well. It doesn't re-optimise values if you have auto optimization on and change the token value in the corner. But that doesn't really matter, as soon as you change an amount it re-optimizes.

evgenyboxer commented on src/components/cards/CreatePool/TokenPrices.vue:
```diff
@@ -0,0 +1,112 @@
+          <BalStack horizontal justify="center">
+            <div>
+              <div class="-mr-1">
+                <span>{{ fNum(priceFor(token), 'usd') }}</span>
```
minor - instead of `usd` you can use `currency` from userSettings

1saf replied:
ye. though I think this is pretty major. good pickup

evgenyboxer commented on src/components/cards/CreatePool/TokenPrices.vue:
```diff
@@ -0,0 +1,112 @@
+        >
+          <BalStack horizontal isDynamic justify="between">
+            <span
+              :class="{ 'font-medium': injectedPrices[token] === undefined }"
```
can this be null? because you can check for both null/undefined using double equal `== null`

1saf replied:
sure

1saf replied:
actually I think I prefer the undefined check here.

1saf replied:
or we could do a falsy check

evgenyboxer commented on src/components/cards/CreatePool/TokenPrices.vue:
```diff
@@ -0,0 +1,112 @@
+              horizontal
+              align="center"
+            >
+              <span>{{ fNum(injectedPrices[token], 'usd') }}</span>
```
same comment about using `currency`

evgenyboxer commented on src/pages/pool/create.vue:
```diff
@@ -325,36 +328,25 @@ watch(hasUnknownToken, () => {
         <BalAccordion
-          :sections="[{ title: 'Pool summary', id: 'pool-summary' }]"
+          :sections="[
+            { title: 'Pool summary', id: 'pool-summary' },
```
seems like this one is missing a translation

evgenyboxer commented on src/components/cards/CreatePool/TokenPrices.vue:
```diff
@@ -0,0 +1,112 @@
+  validTokens.value.filter(t => priceFor(t) === 0 || injectedPrices.value[t])
+);
+const hasUnknownPrice = computed(() =>
+  validTokens.value.some(t => priceFor(t) === 0)
```
I know its a bit annoying, but its easier to read `token` instead of `t` for the array filters.

1saf replied:
yeah good point

1saf commented:
@evgenyboxer  I've addressed all the comments so I'll merge this bad boy into the base branch for unknown tokens. If you have any more suggestions lemme know in the base branch PR :)
